### ISSUE TITLE
Doesn't work with TypeScript 7.0 RC

### ISSUE THREAD

**Issue by movahhedi** [microsoft/vscode-eslint#2195]
ESLint Error log:
```log
2026-06-20 13:59:47.530 [error] An unexpected error occurred:
2026-06-20 13:59:47.530 [error] Error [ERR_PACKAGE_PATH_NOT_EXPORTED]: No "exports" main defined in C:\...\node_modules\.pnpm\@typescript-eslint+typescri_17c9a417c52e85be01d18f8574a67ed2\node_modules\typescript\package.json
    at exportsNotFound (node:internal/modules/esm/resolve:314:10)
    at packageExportsResolve (node:internal/modules/esm/resolve:662:9)
    at resolveExports (node:internal/modules/cjs/loader:692:36)
    at Module._findPath (node:internal/modules/cjs/loader:759:31)
    at Module._resolveFilename (node:internal/modules/cjs/loader:1468:27)
    at wrapResolveFilename (node:internal/modules/cjs/loader:1056:27)
    at defaultResolveImplForCJSLoading (node:internal/modules/cjs/loader:1080:10)
    at resolveForCJSWithHooks (node:internal/modules/cjs/loader:1101:12)
    at Module._load (node:internal/modules/cjs/loader:1269:25)
    at c._load (node:electron/js2c/node_init:2:18095)
```

The `typescript\package.json`:
```json
{
    "name": "typescript",
    "version": "7.0.1-rc",
    "license": "Apache-2.0",
    "author": "Microsoft Corp.",
    "homepage": "https://www.typescriptlang.org/",
    "description": "Preview CLI and JS API for the native TypeScript compiler port",
    "keywords": [
        "TypeScript",
        "Microsoft",
        "compiler",
        "language",
        "javascript",
        "tsgo"
    ],
    "bugs": {
        "url": "https://github.com/microsoft/typescript-go/issues"
    },
    "repository": {
        "type": "git",
        "url": "https://github.com/microsoft/typescript-go.git"
    },
    "publishConfig": {
        "access": "public"
    },
    "type": "module",
    "preferUnplugged": true,
    "engines": {
        "node": ">=16.20.0"
    },
    "files": [
        "bin",
        "lib",
        "dist",
        "vendor",
        "NOTICE.txt"
    ],
    "bin": {
        "tsc": "./bin/tsc"
    },
    "exports": {
        "./package.json": "./package.json",
        "./unstable/sync": "./dist/api/sync/api.js",
        "./unstable/async": "./dist/api/async/api.js",
        "./unstable/fs": "./dist/api/fs.js",
        "./unstable/proto": "./dist/api/proto.js",
        "./unstable/ast": "./dist/ast/index.js",
        "./unstable/ast/is": "./dist/ast/is.js",
        "./unstable/ast/factory": "./dist/ast/factory.generated.js",
        "./unstable/ast/utils": "./dist/ast/utils.js",
        "./unstable/ast/scanner": "./dist/ast/scanner.js",
        "./unstable/ast/visitor": "./dist/ast/visitor.js",
        "./unstable/ast/clone": "./dist/ast/clone.js"
    },
    "imports": {
        "#getExePath": "./lib/getExePath.js",
        "#enums/*": {
            "types": "./dist/enums/*.enum.d.ts",
            "default": "./dist/enums/*.js"
        },
        "#vscode-jsonrpc/node": "./vendor/vscode-jsonrpc/lib/node/main.js"
    },
    "gitHead": "a093a851632f4704ec2a59857da1dccbe6d336a3",
    "optionalDependencies": {
        "@typescript/typescript-win32-x64": "7.0.1-rc",
        "@typescript/typescript-win32-arm64": "7.0.1-rc",
        "@typescript/typescript-linux-x64": "7.0.1-rc",
        "@typescript/typescript-linux-arm": "7.0.1-rc",
        "@typescript/typescript-linux-arm64": "7.0.1-rc",
        "@typescript/typescript-darwin-x64": "7.0.1-rc",
        "@typescript/typescript-darwin-arm64": "7.0.1-rc",
        "@typescript/typescript-aix-ppc64": "7.0.1-rc",
        "@typescript/typescript-freebsd-arm64": "7.0.1-rc",
        "@typescript/typescript-freebsd-x64": "7.0.1-rc",
        "@typescript/typescript-linux-loong64": "7.0.1-rc",
        "@typescript/typescript-linux-mips64el": "7.0.1-rc",
        "@typescript/typescript-linux-ppc64": "7.0.1-rc",
        "@typescript/typescript-linux-riscv64": "7.0.1-rc",
        "@typescript/typescript-linux-s390x": "7.0.1-rc",
        "@typescript/typescript-netbsd-arm64": "7.0.1-rc",
        "@typescript/typescript-netbsd-x64": "7.0.1-rc",
        "@typescript/typescript-openbsd-arm64": "7.0.1-rc",
        "@typescript/typescript-openbsd-x64": "7.0.1-rc",
        "@typescript/typescript-sunos-x64": "7.0.1-rc"
    }
}
```

**Comment by chimurai** [comment]
Bumped into the same issue.

Here are steps to create a reproduction in a vanilla project with eslint / typescript:

```shell
# create new project
npm init -y

# install stable typescript@6 and eslint@10
npm i -D typescript eslint

# eslint init (See below for the configuration)
npx eslint --init

# eslint with TypeScript 6 >> OK
npx eslint .

# upgrade to typescript 7 rc
npm i -D typescript@rc

# eslint with TypeScript 7 rc >> FAIL
npx eslint .
```

Repo with reproduction: https://github.com/chimurai/typescript-rc-eslint-repro

**Comment by dbaeumer** [comment]
@chimurai thanks!

This is unrelated to the vscode-eslint extension since it reproduces solely on the command line as well.

@chimurai can you please report this to the ESLint team. Thanks!!
